File: src/main/java/com/lessmarkup/interfaces/cache/EntityChangeType.java
package com.lessmarkup.interfaces.cache;

public enum EntityChangeType {
    ADDED(1),
    REMOVED(2),
    UPDATED(3);
    
    private final int type;
    
    EntityChangeType(int type) {
        this.type = type;
    }
    
    public int getChangeType() {
        return this.type;
    }
    
    public static EntityChangeType of(int type) {
        switch (type) {
            case 1:
                return ADDED;
            case 2:
                return REMOVED;
            case 3:
                return UPDATED;
            default:
                return null;
        }
    }
}

File: src/main/scala/com/lessmarkup/interfaces/cache/EntityChangeType.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.cache

object EntityChangeType {

  def apply(value: Int) = new EntityChangeType(value)

  final val ADDED = EntityChangeType(1)
  final val REMOVED = EntityChangeType(2)
  final val UPDATED = EntityChangeType(3)
}

class EntityChangeType(val value: Int)
